utils/validate_pr_quickstarts.js

```diff
@@ -0,0 +1,417 @@
+    splitFilePath.findIndex((path) => path === targetChild) - 1;
+
+  let uniqueQuickstartDirectory = splitFilePath[baseQuickstartDirectoryIndex];
+  let indexCounter = baseQuickstartDirectoryIndex;
+
+  while (indexCounter > 1) {
+    uniqueQuickstartDirectory = splitFilePath[indexCounter - 1].concat(
+      `/${uniqueQuickstartDirectory}`
+    );
+    indexCounter--;
+  }
+  return uniqueQuickstartDirectory;
+};
+
+/**
+ * Identifies where in a given file path to look for a quickstart directory.
+ * @param {String} filePath - Full file path of a file in a quickstart.
+ * @return {Function|undefined} Called function with arguments to determine the quickstart of a given file path.
+ */
+const getQuickstartFromFilename = (filePath) => {
+  if (!filePath.includes('quickstarts/')) {
+    return undefined;
+  }
+
+  if (filePath.includes('/alerts/')) {
+    return getQuickstartNode(filePath, 'alerts');
+  }
+
+  if (filePath.includes('/dashboards/')) {
+    return getQuickstartNode(filePath, 'dashboards');
+  }
+
+  if (filePath.includes('/images/')) {
+    return getQuickstartNode(filePath, 'images');
+  }
+
+  const targetChildNode = filePath.split('/').pop();
+
+  return getQuickstartNode(filePath, targetChildNode);
+};
+
+/**
+ * Looks up corresponding quickstart config files for quickstarts known to have changes in a PR.
+ * @param {Set} quickstartDirectories - Set of unique quickstart directories.
+ * @return {Array} Collection of config file paths.
+ */
+const getQuickstartConfigPaths = (quickstartDirectories) => {
+  const allQuickstartConfigPaths = findMainQuickstartConfigFiles();
+
+  return Array.from(quickstartDirectories).reduce(
+    (acc, quickstartDirectory) => {
+      const match = allQuickstartConfigPaths.find((path) =>
+        path.includes(`/${quickstartDirectory}/`)
+      );
+      if (match) {
+        acc.push(match);
+      }
+      return acc;
+    },
+    []
+  );
+};
+
+/**
+ * Builds input argument for submitQuickstart GraphQL mutation.
+ * @param {Object} quickstartConfig - An object containing the path and contents of a quickstart config file.
+ * @return {Object} An object that represents a quickstart in the context of a GraphQL mutation.
+ */
+const buildMutationVariables = (quickstartConfig) => {
+  const {
+    authors,
+    description,
+    title,
+    documentation,
+    logo,
+    keywords,
+    summary,
+    installPlans,
+    id,
+  } = quickstartConfig.contents[0] || {};
+  const alertConfigPaths = getQuickstartAlertsConfigs(quickstartConfig.path);
+  const dashboardConfigPaths = getQuickstartDashboardConfigs(
+    quickstartConfig.path
+  );
+
+  return {
+    id: id ? id : MOCK_UUID,
+    quickstartMetadata: {
+      alertConditions:
+        alertConfigPaths.length > 0
+          ? adaptQuickstartAlertsInput(alertConfigPaths)
+          : undefined,
+      authors: authors && authors.map((author) => ({ name: author })),
+      categoryTerms: getCategoriesFromKeywords(keywords),
```

Comment:
@jerelmiller or @tyreer, does nerdgraph expect us to provide the categories? I vaguely remember talking about this. Initially keywords were associated with particular keywords so that we could update the quickstarts and categories independently of each other. If we are specifying categories explicitely then we could go in and completely remove keywords

Reply:
Related but not an answer per se, just a bit of context buried in a comment thread on this other PR: https://github.com/newrelic/newrelic-quickstarts/pull/665#discussion_r776896725

Reply:
Rereading that thread though, I believe NerdGraph does expect categories. But we're anticipating users would just supply keywords via config, which we map to the categories with this function.

Reply:
Got it! That makes total sense!

Reply:
> does nerdgraph expect us to provide the categories?

**Short answer**: yes

**Long answer**: Since we are designing the API to be accessible to more than just this repo, we made the decision to enforce the requirement that in order to associate a quickstart with a category, it would need to be sent in the mutation as part of the `categoryTerms` field. This makes it obvious to a consumer of the mutation which are actual intended keywords (free text) vs the predefined set of categories (restricted set).

>  Initially keywords were associated with particular keywords so that we could update the quickstarts and categories independently of each other

This is still true, though we call these "category terms". The list of category terms is made up of those list of predefined keywords that we designated as categories. You can see this in NerdGraph using [this query](https://api.newrelic.com/graphiql?#query=%7B%0A%20%20actor%20%7B%0A%20%20%20%20nr1Catalog%20%7B%0A%20%20%20%20%20%20categories%20%7B%0A%20%20%20%20%20%20%20%20displayName%0A%20%20%20%20%20%20%20%20terms%0A%20%20%20%20%20%20%7D%0A%20%20%20%20%7D%0A%20%20%7D%0A%7D%0A). 


> If we are specifying categories explicitely then we could go in and completely remove keywords

I wouldn't completely remove keywords. Keywords are still useful as a free form text field. These keywords are matched when you type something into the search bar when trying to filter down the quickstarts. Keywords are still useful to provide some additional metadata to give you better search results. They just are free-from rather than a predefined set of values.

We don't necessarily need to go update all the quickstart definitions yet, and this is where that `utils/instant-observability-categories.json` file comes into play. Rather than needing to go update the quickstarts right now (introducing a new field to the schema, mass updates on the config files, etc), the idea was that the script could go looks at the configured keywords and pluck the keywords that match the predefined `categoryTerms`. The script would then submit the quickstart with those `categoryTerms` in the mutation. We just need to know which keywords are considered "category terms" in order to compare the keywords and pluck the right values. Make sense?

This buys us time to allow authors to continue to use `keywords`, but maintain the ability to separate the concerns on the API side. I don't see this as a long-term thing, just a stop-gap until we can get a `categoryTerms` field introduced to the yaml configuration. Once we can get a proper field in the config, we can remove this special treatment.

Does that help?